extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -572,7 +573,19 @@ export const popupMessageListener = (request: Request, sessionStore: Store) => {
     const { publicKey } = request;
     await _activatePublicKey({ publicKey });
 
-    sessionStore.dispatch(timeoutAccountAccess());
+    const password = passwordSelector(sessionStore.getState()) || "";
```

Comment:
this is the biggest change here: storing the new account's private key when we change accounts. Just in case something goes wrong in the `try`, we'll still be able to change the account but the user will need to manually re-auth (which is the current UX)